File: 2024/May/5-9 3075 Maximize Happiness of Selected Children/MaxHappiness.py
from typing import List

class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        offset = 0
        result = 0
        for i in range(k):
            if happiness[i] - offset > 0:
                result += happiness[i] - offset
            offset += 1
        
        return result

File: 2024/May/5-9 3075 Maximize Happiness of Selected Children/MaxHappiness.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness);
        int offset = 0;
        long result = 0;
        int index = happiness.length - 1;
        for(int i = 0; i < k; i++) {
            if(happiness[index] - offset > 0) {
                result += happiness[index] - offset;
            }
            offset++;
            index--;
        }

        return result;
    }
}